tests/audiohelper.js

```diff
@@ -471,6 +471,42 @@ describe('AudioHelper', () => {
           });
         });
 
+        describe('inputDevicePromise', () => {
+          [0, null, undefined, {}, false, true, 'non-existent'].forEach((id) => {
+            it(`should reject if id is ${id} and set the internal promise to null`, async () => {
+              audio.setInputDevice(id);
+              const promise = audio._getInputDevicePromise();
+              assert(!!promise);
+              const stub = sinon.stub();
+              try {
+                await promise;
+              } catch {
+                stub();
+              }
+              sinon.assert.calledOnce(stub);
+              assert(!audio._getInputDevicePromise());
+            });
+          });
+
+          it('should resolve and set the internal promise to null', async () => {
+            audio.setInputDevice('input');
+            const promise = audio._getInputDevicePromise();
+            assert(!!promise);
+            await promise;
+            assert(!audio._getInputDevicePromise());
```

Comment:
Perhaps too nitpicky for a unit test, but this will also pass other `falsy` values like `undefined` and `false`, not just `null`.